substrate/primitives/genesis-builder/src/lib.rs

```diff
@@ -49,6 +57,22 @@ sp_api::decl_runtime_apis! {
 		/// It is recommended to log any errors encountered during the process.
 		///
 		/// Please note that provided json blob must contain all `GenesisConfig` fields, no defaults will be used.
-		fn build_config(json: sp_std::vec::Vec<u8>) -> Result;
+		#[renamed("build_config", 2)]
+		fn build_state(json: sp_std::vec::Vec<u8>) -> Result;
+
+		/// Returns a JSON blob representation of the builtin `GenesisConfig` identified by `id`.
+		///
+		/// If `id` is `None` the function instantiates the default `GenesisConfig` struct for the runtime and serializes it into a JSON
+		/// blob. It returns a `Vec<u8>` containing the JSON representation of the default `GenesisConfig`.
+		/// Otherwise returns a JSON representation of the built-in, named, pre-configured `GenesisConfig` value.
+		#[api_version(2)]
+		fn get_preset(id: Option<sp_std::vec::Vec<u8>>) -> Option<sp_std::vec::Vec<u8>>;
+
+		/// Returns a list of available builtin `GenesisConfig` presets.
+		///
+		/// The presets from the list can be queried with [`GenesisBuilder::get_preset`] method. If no named presets are
+		/// provided by the runtime the list is empty.
+		#[api_version(2)]
+		fn preset_names() -> sp_std::vec::Vec<sp_runtime::RuntimeString>;
```

Comment:
@bkchr I'd appreciate feedback on naming. `config` is not advised (see #150), so maybe `preset` will do the job?

I am also not sure if we need to keep `create_default_config` method. Maybe we can remove it?

Reply:
>I am also not sure if we need to keep create_default_config method. Maybe we can remove it?

I am currently using that in the new chain-agnostic runtime benchmarker (https://github.com/paritytech/polkadot-sdk/pull/3512), how would i know what preset is the default?

Reply:
`get_preset(None)` would return default.

Reply:
Did some cleanup here. 

I dropped old API and introduced new functions as I assume the node will not support old runtimes when it comes to spec generating.

Introduced `PresetId`.

Reply:
Did some cleanup here. 

I dropped old API and introduced new functions as I assume the node will not support old runtimes when it comes to spec generating.